extensions/amp-analytics/0.1/crc32.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * Standard key for CRC32
+ * https://en.wikipedia.org/wiki/Cyclic_redundancy_check#Polynomial_representations_of_cyclic_redundancy_checks
+ * @const {number}
+ */
+const CRC32_KEY = 0xEDB88320;
+
+/** @const {Array<number>} */
+const CRC_TABLE = makeCrcTable();
```

Comment:
can we lazily initialize the table? we should not do this work at runtime start up time

Reply:
Done.